### ISSUE TITLE
Check for not defined ref on a component when using script setup

### ISSUE THREAD

**Issue by Demivan** [vuejs/eslint-plugin-vue#2104]
**Please describe what the rule should do:**

When using script setup check that ref used on a component was defined.

This will protect against typos of component ref attribute.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

Good:
```vue
<script setup>
const definedRef = ref()
</script>

<template>
  <div ref="definedRef"></div>
</template>
```

Bad:
```vue
<script setup>
const definedRef = ref()
</script>

<template>
  <div ref="notDefinedRef"></div>
                   ^ error: notDefinedRef is not defined
</template>
```

**Additional context**

**Comment by ota-meshi** [comment]
Thank you for the rule suggestions.
But I think you can check for the same using the no-unused-refs rule.

https://eslint.vuejs.org/rules/no-unused-refs.html

[DEMO](https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxtjMEKwyAQRH9F9tRCgtBjMDn1C3r2UnQFIV1F11xC/r0rhRRKjzPvzezgkkeYwFRXYmZVkVteLLlElZXHEAn9A4OaVcFwuVoy+qOKJIHxldcnoySljI9b12YLlPh+ji0sRgv7lfw/w+jzEwYobcUK0w5bQ01pbCSjMZeUsXDs6DZ8WavoR/nu9XG8AXOpTQI=)
What do you think?

**Comment by Demivan** [comment]
My bad, missed it somehow. That is exactly what I need. Thanks.
Curious why it is not in recommended rules or even essential? Any downside in adding it to those categories?

**Comment by ota-meshi** [comment]
Changing a shareable config is a breaking change.
Also, when we add new rules to the config, users complain that the rules have become stricter. So anymore I don't want to add new rules to the config.

**Comment by Demivan** [comment]
Makes sense, thanks for explaining. Will close this now.
